src/scripts/api-service.js

```diff
@@ -8,8 +8,8 @@ import {
 
 import { flightData$ } from "./observables";
 import { mapObserver } from "./map-service";
-import { switchMap } from "rxjs/operators";
 import { listObserver } from "./dom-manip";
+
 async function getAllStates() {
 	if (!(await isRequestInCache(`${baseUrl}/states/all?extended=1`))) {
```

Comment:
cache.match(requestedUrl) in your web storage files returns a promise . Use the from operator to convert it to an observable, then handle the logic once subscribed.

Reply:
Resolved issue, please check if fix is aknowledged.